aws_lambda_powertools/utilities/idempotency/base.py

```diff
@@ -138,7 +139,7 @@ def _get_remaining_time_in_millis(self) -> Optional[int]:
 
         return None
 
-    def _get_idempotency_record(self) -> DataRecord:
+    def _get_idempotency_record(self) -> Optional[DataRecord]:
```

Comment:
@heitorlessa Instead of creating an empty DataRecord in the `get_record` method, I change the return to be optional and then I can return None.